src/CloudFoundry.VisualStudio/MSBuild/MSBuildProcess.cs

```diff
@@ -0,0 +1,105 @@
+﻿namespace CloudFoundry.VisualStudio.MSBuild
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Build.Evaluation;
+    using EnvDTE;
+    using Microsoft.VisualStudio.Shell;
+ 
+    public class MSBuildProcess : IMSBuildProcess
+    {
+        public void Push(string projectFullPath, string pubxml, string logFile, Microsoft.Build.Framework.LoggerVerbosity verbosity)
+        {
+            System.Threading.Tasks.Task.Factory.StartNew(() =>
+            {
+                DTE dte = (DTE)CloudFoundry_VisualStudioPackage.GetGlobalService(typeof(DTE));
+                var window = dte.Windows.Item(EnvDTE.Constants.vsWindowKindOutput);
+                var output = (OutputWindow)window.Object;
+                bool paneFound = false;
+
+                OutputWindowPane pane = null;
+
+                try
+                {
+                    foreach (OutputWindowPane item in output.OutputWindowPanes)
+                    {
+                        if (item.Name == "Publish")
+                        {
+                            pane = output.OutputWindowPanes.Item("Publish");
+                            paneFound = true;
+                            break;
+                         }
+                    }
+
+                    if (paneFound == false)
+                    {
+                        pane = output.OutputWindowPanes.Add("Publish");
+                    }
+                    pane.Activate();
+                }
+                catch (Exception ex)
+                {
+                    throw ex;
+                }
+
+                
+                var engine = InitializeMsBuildEngine(null);
+
+                ErrorListProvider errorList = CloudFoundry_VisualStudioPackage.GetErrorList();
+
+                if (errorList == null)
+                {
+                    throw new InvalidOperationException("Could not retrieve error list provider");
+                }
+                
+                errorList.Tasks.Clear();
+
+                Microsoft.Build.Framework.ILogger fileLogger = new Microsoft.Build.Logging.FileLogger();
+
+                MSBuildLogger customLogger = new MSBuildLogger(pane, errorList);
+
+                customLogger.Verbosity = verbosity;
+
+                fileLogger.Parameters = string.Format("logfile={0}", logFile);
+
+                fileLogger.Verbosity = verbosity;
+
+                engine.RegisterLogger(fileLogger);
+                engine.RegisterLogger(customLogger);
+
+                engine.SetGlobalProperty("DeployOnBuild", "true");
```

Comment:
Instead of hard-coding these in `MSBuildProcess` class, can we expose a [`StringDictionary`](https://msdn.microsoft.com/en-us/library/system.collections.specialized.stringdictionary%28v=vs.110%29.aspx) property instead?